api/src/pcapi/core/providers/titelive_book_search.py

```diff
@@ -0,0 +1,249 @@
+        }
+
+    def deserialize_titelive_search_json(
+        self, titelive_json_response: dict[str, typing.Any]
+    ) -> TiteliveProductSearchResponse[TiteLiveBookOeuvre]:
+        return pydantic.parse_obj_as(TiteliveProductSearchResponse[TiteLiveBookOeuvre], titelive_json_response)
+
+    def filter_allowed_products(
+        self, titelive_product_page: TiteliveProductSearchResponse[TiteLiveBookOeuvre]
+    ) -> TiteliveProductSearchResponse[TiteLiveBookOeuvre]:
+        for oeuvre in titelive_product_page.result:
+            oeuvre.article = [
+                article for article in oeuvre.article if self.is_book_article_allowed(article, oeuvre.titre)
+            ]
+        return titelive_product_page
+
+    def get_not_allowed_eans(
+        self, titelive_product_page: TiteliveProductSearchResponse[TiteLiveBookOeuvre]
+    ) -> list[str]:
+        not_cgu_compliant_eans = []
+        for oeuvre in titelive_product_page.result:
+            not_cgu_compliant_eans += [
+                article.gencod for article in oeuvre.article if not self.is_book_article_allowed(article, oeuvre.titre)
+            ]
+        return not_cgu_compliant_eans
+
+    def upsert_titelive_result_in_dict(
+        self, titelive_search_result: TiteLiveBookOeuvre, products_by_ean: dict[str, offers_models.Product]
+    ) -> dict[str, offers_models.Product]:
+        title = titelive_search_result.titre
+        authors = titelive_search_result.auteurs_multi
+        for article in titelive_search_result.article:
+            ean = article.gencod
+            product = products_by_ean.get(ean)
+            if product is None:
+                products_by_ean[ean] = self.create_product(article, title, authors)
+            else:
+                products_by_ean[ean] = self.update_product(article, title, authors, product)
+        return products_by_ean
+
+    def create_product(self, article: TiteLiveBookArticle, title: str, authors: list[str]) -> offers_models.Product:
+        return offers_models.Product(
+            description=article.resume,
+            extraData=build_book_extra_data(article, authors),
+            idAtProviders=article.gencod,
+            lastProvider=self.provider,
+            name=title,
+            subcategoryId=LIVRE_PAPIER.id,
+        )
+
+    def update_product(
+        self,
+        article: TiteLiveBookArticle,
+        title: str,
+        authors: list[str],
+        product: offers_models.Product,
+    ) -> offers_models.Product:
+        ean = article.gencod
+        product.description = article.resume
+        if product.extraData is None:
+            product.extraData = offers_models.OfferExtraData()
+        product.extraData.update(build_book_extra_data(article, authors))
+        product.idAtProviders = ean
+        product.name = title
+        product.subcategoryId = LIVRE_PAPIER.id
+
+        # If the product is updated to eligible, it is because the offers must be approved to become ineligible due to gcu
```

Comment:
[question] Je ne comprends pas la phrase, car je ne comprends pas la fin : "the offers must be approved to become ineligible".  Si une offre est approuvée, elle ne peut pas être inéligible, non ? Par définition, une offre approuvée est forcément éligible, non ?

(J'ai bien vu que ce commentaire est copié du code actuel.)

Reply:
Oui
Je cromprend pas non plus. 
Je retire ce commentaire plus confusant qu'aidant